### ISSUE TITLE
Auto-select the output_device

### ISSUE THREAD

**Issue by jbonlinea** [mikebrady/shairport-sync#493]
Hi there, 

I installed Shareport-sync on my Rpi (latest raspbian) following your explanaitions, and it works like à charm ! Many thank's.
I'm now wondering if there is a way to tell/force Shaireport-sync to use the raspberry `curent default soud-card` as its `output_device` ?

Let me explain myself.
On my Rpi, I have three audio card : 
- 0 : the built in card, 
- 1 : an Hifibery Dac+, 
- 2 : a cheapo USB wireless sound card (UBS is transmitter and there is a receiver/DAC with stereo rca outputs)

All works well with the Rpi and I can tell Shaireport-sync to use one or the other sound card by editing the `alsa` section of its config file in `/etc/shairport-sync.conf`. Great !

However, my soud cards are not plugged in the same ampfire, 
-  card 1 is plugged into my primary Hifi in my living room, 
-  card 2 is plugge into a secondary Hifi in my kitchen.

From the Rpi UI, i can for instance listen to a web radio and select my output device by changing the "default sound card" from the "audio device setting".

Now I would like that the `output_device` of Shareport-sync change as I change my "default sound card" from the "audio device setting".

I tried to edit `/usr/share/alsa/alsa.conf` and change `pcm.front cards.pcm.front` to `pcm.front cards.pcm.default` but it didn't help.
Well maybe it was a stupid try, I'm kind of a linux noob.

This would be super handy ! (not to say tha that having to edit the config file and reboot the Rpi is quite painfull for a day to day use), however I realise this might be tricky. 

Indeed, if I change the `default audio sound card` from the `audio device setting`, I still have to restart my web broswer for it to use the `newly set default sound card` otherwise it keeps using the one that was set to default when it was launched. 

But yet, that would be awesome !

Anyhow awesome app, thank's !

**Comment by mikebrady** [comment]
Thanks for the kind words.

Actually, what you ask would be difficult – maybe impossible in general – because sometimes the default output device is not a real hardware device (which Shairport Sync needs), but may be a PulseAudio device or an ALSA virtual device doing things like rate changing and other stuff. Generally, these non-real devices do not report their latencies or delays properly and are unsuitable for use by Shairport Sync.

**Comment by jbonlinea** [comment]
Hum, I see... then this was no more than a wishfull thinking :-)

Thank's for your quick reply !

ps : as you might be much more informed than I'm, I'm looking for a similar/as efficient solution than Shaireport-sync (no really I'm impressed how well it works) but :
- for Chromecast (or any android device)
- in order to turn my Rpi into a WiFi audio receiver (not "protocol" based)
- to turn my raspi into a bluetooth audio receiver, so I can stream any bleutooth device through my cool Hifiberry DAC+
- to set up a multiroom system based on raspbian (and not alternative distribution like PiCorePlayer, Max2play, or PiMusicBox) ; of couse I'm willing to install a squezeebox server or any relevent alternative

Well, ideally I would like one Rpi to be able to do all that, and use more Rpi as "simple" receiver

Thank's in advance

**Comment by mikebrady** [comment]
Yeah, I'm afraid it does seem to be too tricky to implement well.

Regarding other audio apps, I'm, not really terribly well informed at all, though people seem to be looking at SnapCast as well as Shairport Sync.

**Comment by jbonlinea** [comment]
No problem ! 
Shareport-syn is by far the best audio streaming software I've found ! 

I'll have a look at snapcast, never heard about it 

Thank's again
